matRad_rayTracing.m

```diff
@@ -0,0 +1,165 @@
+
+for j = 1:stf.numOfRays
+
+    tmp_rayMx_bev_x = repmat(round(stf.ray(j).rayPos_bev(1) / rayMxSpacing) + [-numOfRayTracingRays:numOfRayTracingRays],2*numOfRayTracingRays+1,1);
+    tmp_rayMx_bev_y = rayMx_bev_z * ones(2*numOfRayTracingRays+1);
+    tmp_rayMx_bev_z = repmat(round(stf.ray(j).rayPos_bev(3) / rayMxSpacing) + [-numOfRayTracingRays:numOfRayTracingRays]',1,2*numOfRayTracingRays+1);
+
+    rayMx_bev = [rayMx_bev; [rayMxSpacing * tmp_rayMx_bev_x(:) tmp_rayMx_bev_y(:) rayMxSpacing * tmp_rayMx_bev_z(:)]];
+
+    rayMx_bev = unique(rayMx_bev,'rows');
+
+end
+
+%     figure,
+%     for jj = 1:length(rayMx_bev)
+%        plot(rayMx_bev(jj,1),rayMx_bev(jj,3),'rx'),hold on 
+%     end
+    
+% Rotation around Z axis (gantry)
+rotMx_XY_T = [ cosd(stf.gantryAngle) sind(stf.gantryAngle) 0;
+              -sind(stf.gantryAngle) cosd(stf.gantryAngle) 0;
+                                   0                     0 1];
+    
+% Rotation around Y axis (couch)
+rotMx_XZ_T = [cosd(stf.couchAngle) 0 -sind(stf.couchAngle);
+                                 0 1                     0;
+              sind(stf.couchAngle) 0  cosd(stf.couchAngle)];
+
+% rotate ray matrix from bev to world coordinates
+rayMx_world = rayMx_bev * rotMx_XY_T * rotMx_XZ_T;
+
+% set up distance cube to decide which rad depths should be stored
+rayTracingDotProdCube = -inf*ones(size(ct.cube));
+
+% perform ray tracing over all rays
+for j = 1:size(rayMx_world,1)
+
+    % run siddon ray tracing algorithm
+    [alphas,l,rho,d12,ixHitVoxel] = matRad_siddonRayTracer(stf.isoCenter, ...
+                                ct.resolution, ...
+                                stf.sourcePoint, ...
+                                rayMx_world(j,:), ...
+                                {ct.cube});
+                                                        
+    % find voxels for which we should remember this tracing because this is
+    % the closest ray
+    normRayVector = rayMx_world(j,:) - stf.sourcePoint;
+    normRayVector = normRayVector/norm(normRayVector);
+
+    dotProdHitVoxels = coords(ixHitVoxel,:)*normRayVector'; % this also corresponds to the geometrical distance!!!
+
+    ixRememberFromCurrTracing = dotProdHitVoxels > rayTracingDotProdCube(ixHitVoxel)';
+
+    if sum(ixRememberFromCurrTracing) > 0
+        rayTracingDotProdCube(ixHitVoxel(ixRememberFromCurrTracing)) = dotProdHitVoxels(ixRememberFromCurrTracing);
+
+        % calc radioloical depths
+
+        % eq 14
+        % It multiply voxel intersections with \rho values.
+        % The zero it is neccessary for stability purpose.
+        d = [0 l .* rho{1}]; %Note. It is not a number "one"; it is the letter "l"
+
+        % Calculate accumulated d sum.
+        dCum = cumsum(d);
+
+        % Calculate the radiological path
+        vRadDepth = interp1(alphas,dCum,dotProdHitVoxels(ixRememberFromCurrTracing)/d12,'linear');
```

Comment:
We need the extrapolation value zero.

And then the following sanity check is useless.